waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -86,12 +97,26 @@ constructAppSpec waspDir options decls = do
     [] -> Right appSpec
     validationErrors -> Left $ map show validationErrors
 
-findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Either String (Path' Abs File'))
+findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
 findWaspFile waspDir = do
   files <- fst <$> IOUtil.listDirectory waspDir
-  return $ maybeToEither "Couldn't find a single *.wasp file." $ (waspDir </>) <$> find isWaspFile files
+  return $ (waspDir </>) <$> find isWaspFile files
   where
     isWaspFile path =
       ".wasp"
         `isSuffixOf` toFilePath path
         && (length (toFilePath path) > length (".wasp" :: String))
+
+analyzePackageJsonContent :: Path' Abs (Dir WaspProjectDir) -> IO (Either [CompileError] PackageJson)
+analyzePackageJsonContent waspProjectDir =
+  findPackageJsonFile >>= \case
+    Just packageJsonFile -> readPackageJsonFile packageJsonFile
+    Nothing -> return $ Left [fileNotFoundMessage]
+  where
+    fileNotFoundMessage = "couldn't find package.json file in the " ++ toFilePath waspProjectDir ++ " directory"
+    findPackageJsonFile = findFileInWaspProjectDir waspProjectDir [relfile|package.json|]
```

Comment:
So this path to package.json we have here, how about we take that a bit higher up somewhere, so others can also access it if needed? Feels like it should be global knowledge, same like position of env files or dockerfile, but it is embedded here into `where`.

Reply:
I agree, I'll move it.